crates/contracts/src/alloy.rs

```diff
@@ -985,5 +994,79 @@ mod tests {
             assert!(UniswapV2Factory::deployment_address(chain_id).is_some());
             assert!(UniswapV2Router02::deployment_address(chain_id).is_some());
         }
+
+        for chain_id in &[
+            MAINNET,
+            ARBITRUM_ONE,
+            BASE,
+            AVALANCHE,
+            BNB,
+            OPTIMISM,
+            POLYGON,
+            LENS,
+            LINEA,
+            PLASMA,
+        ] {
+            assert!(UniswapV3QuoterV2::deployment_address(chain_id).is_some());
+        }
+
+        for chain_id in &[
+            MAINNET,
+            ARBITRUM_ONE,
+            BASE,
+            AVALANCHE,
+            BNB,
+            OPTIMISM,
+            POLYGON,
+            LENS,
+            LINEA,
+            PLASMA,
+        ] {
+            assert!(UniswapV3SwapRouterV2::deployment_address(chain_id).is_some());
+        }
+
+        for chain_id in &[
+            MAINNET,
+            SEPOLIA,
+            ARBITRUM_ONE,
+            BASE,
+            AVALANCHE,
+            BNB,
+            OPTIMISM,
+            POLYGON,
+            LENS,
+            LINEA,
+            PLASMA,
+        ] {
+            assert!(IUniswapV3Factory::deployment_address(chain_id).is_some());
+        }
+
+        for chain_id in &[
+            MAINNET,
+            GNOSIS,
+            SEPOLIA,
+            ARBITRUM_ONE,
+            BASE,
+            AVALANCHE,
+            BNB,
+            OPTIMISM,
+            POLYGON,
+            PLASMA,
+        ] {
+            assert!(Permit2::deployment_address(chain_id).is_some());
+        }
+
+        for chain_id in &[
+            MAINNET,
+            GNOSIS,
+            SEPOLIA,
+            ARBITRUM_ONE,
+            BASE,
+            AVALANCHE,
+            OPTIMISM,
+            PLASMA,
+        ] {
+            assert!(BalancerV3BatchRouter::deployment_address(chain_id).is_some());
+        }
```

Comment:
Nit: these are being removed by #3827 (https://github.com/cowprotocol/services/pull/3827/files#diff-42edfcaf70e931c1a5d1090283d4b0be752f7267cf7c288cd9c0c663bb449bd3)

I can remove them after though, nw

Reply:
Quick look from you, and it should be good to merge ser 🤩